### ISSUE TITLE
[BUG] {{ Two data links returned when accessing AU_SI12_NRT_R04 }}

### ISSUE THREAD

**Issue by asteiker** [nsidc/earthaccess#859]
### Is this issue already tracked somewhere, or is this a new report?

- [X] I've reviewed existing issues and couldn't find a duplicate for this problem.

### Current Behavior

Reported by @trey-stafford in separate issue under #307: For AU_SI12_NRT_R04, the granules have two data links which have the same data. One is pretty fast but the other link is slow to download.

```
results[-1].data_links()
[
    'https://lance.nsstc.nasa.gov/amsr2-science/data/level3/seaice12/R04/hdfeos5/AMSR_U2_L3_SeaIce12km_R04_20241016.he5', 
    'https://lance.itsc.uah.edu/amsr2-science/data/level3/seaice12/R04/hdfeos5/AMSR_U2_L3_SeaIce12km_R04_20241016.he5'
]
```

This seems like an issue with the CMR metadata itself, but I don't recall how earthaccess is handling multiple get data URLs and whether there should be better behavior to only utilize one of the links. 

### Expected Behavior

Only a single link per granule should be used for opening or downloading granules from earthaccess instead of duplicating the outputs.

### Steps To Reproduce

Copying original sample code from #307:

```
import earthaccess

results = earthaccess.search_data(short_name='AU_SI12_NRT_R04')
results = sorted(results, key=lambda x: x['meta']['revision-date'], reverse=True)
earthaccess.login()
files = earthaccess.download(results, "/tmp/test")
```

### Environment

```markdown
Using earthaccess v0.11.0
```

**Comment by arthur-e** [comment]
I am having the same issue for a different asset. The same granule is available as two different URLs.

```python
year = 2019
granules = earthaccess.search_data(
    short_name = 'spl3ftp_e', version = '004',
     temporal = (f'{year}-01-01', f'{year}-12-31'))
```
```
# Granules includes, for instance:
[
    'https://data.nsidc.earthdatacloud.nasa.gov/nsidc-cumulus-prod-protected/SMAP/SPL3FTP_E/004/2019/12/SMAP_L3_FT_P_E_20191201_R19240_001.h5', 
    'https://n5eil01u.ecs.nsidc.org/DP4/SMAP/SPL3FTP_E.004/2019.12.01/SMAP_L3_FT_P_E_20191201_R19240_001.h5'
]
```

**Comment by asteiker** [comment]
@arthur-e Thank you for reporting this. Granted I am testing with `earthaccess` version `0.12.0`, but I believe this may be a separate issue. As of December 2024, SMAP is now available in the Earthdata Cloud, as well as its original archive location on-premises. Those two different URLs you see in your SPL3FTP_E example are for each location. But, these are actually two distinct data products as far how they are reflected in the Common Metadata Repository (CMR), which is what `earthaccess` uses behind the scenes. If you include `cloud-hosted = True` in your search_data query, then you will only receive the URLs, and associated granules, from Earthdata Cloud. We recommend including this in your search to avoid these duplicates, and to become familiar with the cloud-hosted URLs, as we will be decommissioning the on-premises URLs in the near future.

**Comment by arthur-e** [comment]
@asteiker Great, thanks! I assume `cloud_hosted = True` just reflects the datasets that have the "Dataset is available in the Earthdata Cloud" icon in Earthdata Search.

**Comment by asteiker** [comment]
@arthur-e Yes that is true, the `cloud_hosted = True` parameter is searching against the same filter as you see in Earthdata Search. However, because the NSIDC DAAC currently provides both the cloud-hosted and on-premises SMAP data sets, you will actually get duplicate results if you do not specify or filter for the cloud-hosted data. I apologize for any confusion on this! Please let us know if we can make this clearer within the `earthaccess` library.

**Comment by trey-stafford** [comment]
Confirming that the issue remains for datasets that are exposed both through ECS and the cloud:

```
import earthaccess

earthaccess.login()

granules = earthaccess.search_data(
    short_name="ILATM1B",
    version="2",
    granule_name="ILATM1B_20191120_041100.ATM6AT6.h5",
)

for granule in granules:
    print(granule.data_links())

```
```
['https://data.nsidc.earthdatacloud.nasa.gov/nsidc-cumulus-prod-protected/ICEBRIDGE/ILATM1B/2/2019/11/20/ILATM1B_20191120_041100.ATM6AT6.h5']
['https://n5eil01u.ecs.nsidc.org/DP4/ICEBRIDGE/ILATM1B.002/2019.11.20/ILATM1B_20191120_041100.ATM6AT6.h5']
```

Users have to know to include the `cloud_hosted` kwarg when using `search_data`. 

This can cause issues when downloading data. The best case scenario is that the data is downloaded twice, the second download replacing the first. I think I have observed behavior where two threads are competing to download chunks to the same file, resulting in bad data, however. If I can manage to replicate, I will post an example. 

 In the example above, if you download with e.g., `earthaccess.download(granules, "/tmp/test/")`, and then watch the file written to `/tmp/test`, you will see the file grow, then shrink to 0 bytes, and then the second download will take over and write out the file again.

**Comment by itcarroll** [comment]
Thanks for the update. I want to note this is relevant to #565.

**Comment by asteiker** [comment]
Since all NSIDC DAAC data migrated to Earthdata Cloud, with few remaining on-premises across NASA Earthdata, and given the related issue mentioned above, I am going to close this as OBE. Please let me know @trey-stafford or @arthur-e if you have concerns about this or if there is still something additional we ought to fix within the earthaccess library.
